lib/uri.mli

```diff
@@ -0,0 +1,24 @@
+val get_sld : string -> string option
+(** Get the URI's second level domain, if it has one. *)
+
+val append_to_base : rel_path:string -> string -> string
+(** Append a relative path to a base URI. *)
+
+val chop_git_prefix : string -> string
+(** Chop the prefix [git+] from a URI, if present. *)
+
+val to_https : string -> string
+(** Convert [git@] and [git+ssh://] into https URI's. Leave other URI's as is. *)
+
+val to_github_standard : string -> (string, [> Bos_setup.R.msg ]) result
```

Comment:
I think this and the next two functions should probably live a `Github` submodule so we can clarify that all those functions only apply to github URIs.

In the future we can extract  type and use a better abstraction than strings or `Uri.t` for those.

Reply:
> I think this and the next two functions should probably live a Github submodule so we can clarify that all those functions only apply to github URIs.

Yes, I agree, that's better.

> In the future we can extract type and use a better abstraction than strings or Uri.t for those.

That's what @gpetiot did in #215, right? I liked that PR.

Reply:
Yep! That's definitely something we should aim at but I think this PR is already a huge improvement as it is.

Once we're heading towards 2.0 it'll be clearer how to refactor these and also hopefully much simpler since we won't have to parse github.io URIs for instance!